src/Microsoft.AspNetCore.Http.Abstractions/Internal/ParsingHelpers.cs

```diff
@@ -26,7 +26,7 @@ private static IEnumerable<string> GetHeaderSplitImplementation(StringValues val
         {
             foreach (var segment in new HeaderSegmentCollection(values))
             {
-                if (segment.Data.HasValue)
+                if (segment.Data.HasValue && !string.IsNullOrWhiteSpace(segment.Data.Value))
```

Comment:
And yes, I'm working on adding a bunch of methods to StringSegment: https://github.com/aspnet/Common/issues/213